Packages/ClientRuntime/Sources/Waiters/Waiter.swift

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// An object used to wait on an operation.
+/// Intended to be a generic type for use when waiting on any Smithy `operation`.
+/// May be reused for multiple waits, including concurrent operations.
+public class Waiter<Input, Output> {
+
+    /// The configuration this waiter was created with.
+    public let config: WaiterConfig<Input, Output>
+
+    /// The operation that this waiter will call one or more times while waiting on the success condition.
+    public let operation: (Input) async throws -> Output
```

Comment:
why is operation tied to the Waiter itself? This means customer (even AWS SDK) have to use separate waiter for each operation.

Also saving a method signature seems like a quick solution but what would happen if add an extra optional param in the operation?